trpcultivate_phenotypes/tests/src/Kernel/Validators/MetadataInputTest.php

```diff
@@ -0,0 +1,441 @@
+<?php
+
+/**
+ * @file
+ * Kernel tests for validator plugins specific to validating metadata/input to importer.
+ */
+
+namespace Drupal\Tests\trpcultivate_phenotypes\Kernel\Validators;
+
+use Drupal\Tests\tripal_chado\Kernel\ChadoTestKernelBase;
+use Drupal\Tests\trpcultivate_phenotypes\Traits\PhenotypeImporterTestTrait;
+use Drupal\Core\Form\FormState;
+
+ /**
+  * Tests Tripal Cultivate Phenotypes Metadata Validator Plugins.
+  *
+  * @group trpcultivate_phenotypes
+  * @group validators
+  */
+class MetadataInputTest extends ChadoTestKernelBase {
+  use PhenotypeImporterTestTrait;
+
+  /**
+   * A genus that exists and is configured. 
+   */
+  protected $test_genus;
+
+   /**
+   * A project that exists. 
+   */
+  protected $test_project;
+
+  /**
+   * Modules to enable.
+   */
+  protected static $modules = [
+    'user',
+    'tripal',
+    'tripal_chado',
+    'trpcultivate_phenotypes'
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+
+    // Set test environment.
+    \Drupal::state()->set('is_a_test_environment', TRUE);
+
+    // Install module configuration.
+    $this->installConfig(['trpcultivate_phenotypes']);
+
+    // Test Chado database.
+    // Create a test chado instance and then set it in the container for use by our service.
+    $this->connection = $this->createTestSchema(ChadoTestKernelBase::PREPARE_TEST_CHADO);
+    $this->container->set('tripal_chado.database', $this->connection);
+
+    // Set plugin manager service.
+    $this->plugin_manager = \Drupal::service('plugin.manager.trpcultivate_validator');
+
+    $genus = 'Tripalus';
+    // Create our organism and configure it.
+    $organism_id = $this->connection->insert('1:organism')
+      ->fields([
+        'genus' => $genus,
+        'species' => 'databasica',
+      ])
+      ->execute();
+
+    $this->assertIsNumeric($organism_id, 'We were not able to create an organism for testing (configured).');
+    $this->test_genus['configured'] = $genus;
+    $this->setOntologyConfig($this->test_genus['configured']);
+
+    // Create another organism and not configure.
+    $genus = 'notconfiggenus';
+    $organism_id = $this->connection->insert('1:organism')
+      ->fields([
+        'genus' => $genus,
+        'species' => 'databasica',
+      ])
+    ->execute();
+
+    $this->assertIsNumeric($organism_id, 'We were not able to create an organism for testing (not configured).');
+    $this->test_genus['not-configured'] = $genus;
+    
+    // Set terms configuration.
+    $this->setTermConfig();
+
+
+    // Create test project.
```

Comment:
> Lastly, I didn't see any automated test for Lacey's last suggestion:
> 
> > where the project has a genus property that was added without the trpcultivate_phenotypes.genus_project service.
> 

This was added by setting up the project-genus-not-tru-service project by inserting the genus directly to the projectprop table instead of using the service in the setup() ✅